integration/hibernate-base/src/main/java/com/blazebit/persistence/impl/hibernate/HibernateJpaProvider.java

```diff
@@ -502,12 +502,19 @@ protected String getMappedBy(CollectionPersister persister) {
         }
 
         // In this case, the property might represent a formula
-        if (columnNames.length == 1 && columnNames[0] == null) {
+        boolean isFormula = columnNames.length == 1 && columnNames[0] == null;
+        boolean isSubselect = entityPersister.getTableName().contains("(");
+
+        if (isFormula || isSubselect) {
             Type propertyType = entityPersister.getPropertyType(attributeName);
             long length;
             int precision;
             int scale;
             try {
+                if (propertyType instanceof org.hibernate.type.EntityType) {
+                    propertyType = ((org.hibernate.type.EntityType) propertyType).getIdentifierOrUniqueKeyType(sfi);
```

Comment:
This was required to get the `PrimaryKeyJoinColumn` from the reproducer test case to work. I am pretty sure that the column type for an `EntityType` indeed should be the type of its identifying type, but I am uncertain whether this approach is always correct.